src/services/factories/tokenList.ts

```diff
@@ -266,8 +266,12 @@ export function getTokensFactory(factoryParams: {
     }
   }
 
-  async function updateTokenDetails(networkId: number, numTokens: number): Promise<void> {
-    const tokensConfig = tokenListApi.getTokens(networkId)
+  async function updateTokenDetails(
+    networkId: number,
+    numTokens: number,
+    tokensConfig?: TokenDetails[],
+  ): Promise<void> {
+    tokensConfig = (tokensConfig?.length && tokensConfig) || tokenListApi.getTokens(networkId)
```

Comment:
```suggestion
   if (tokensConfig.length === 0) tokensConfig = tokenListApi.getTokens(networkId)
```
maybe

Reply:
with default assignment = [] in args yeah good idea